### ISSUE TITLE
defunct ffmpeg processes left even after clip.close()

### ISSUE THREAD

**Issue by TejasBob** [Zulko/moviepy#745]
I tried following snippets in ipython terminal.

**Snippet 1**:

```
from moviepy.editor import *

clip = VideoFileClip(_video_path_).subclip(10,20).resize((200,100))

clip.write_videofile("result.mov", codec = "libx264", fps= 25)

clip.close()
```

After doing clip.close(),  I checked 
```
$ps -ef | grep ffmpeg
501 56157   343   0 12:55PM ttys000    0:00.00 grep ffmpeg
501 56134 56091   0 12:52PM ttys001    0:00.00 (ffmpeg)
```
There is one ffmpeg process left in memory.
I thought this happened because of loss of reference to ffmpeg process due to 2 outplace operations (subclip and resize), so I tried following script.

**Snippet2**:

```
from moviepy.editor import *
proc_list = []
clip = VideoFileClip(_video_path_)
proc_list.append(clip)

clip = clip.subclip(10,20)
proc_list.append(clip)

clip = clip.resize((200,100))
proc_list.append(clip)

clip.write_videofile("result.mov", codec = "libx264", fps= 25)

for ele in proc_list:
    ele.close()
```

Again I did,
`ps -ef | grep ffmpeg`

It gave similar output. There is one defunct process left in memory.

What is the right way of removing the ffmpeg instances?

Thanks!

**Comment by tburrows13** [comment]
This is an problem with the main branch, that doesn't have a fix yet. It is discussed in #596. There is a fix in #608 that you can try using.

Hope this helps, feel free to ask again if you need more assistance.

**Comment by TejasBob** [comment]
Thank you for references.

I noticed that, defunct ffmpeg processes are left only when the video being read has an audio associated with it. I managed to figure out that subclip() on the audio was causing the problem. 

I added outplace decorator to subclip method in Clip.py and then it stopped leaving zombie processes behind.

Original Clip.py:
```
@convert_to_seconds(['t_start', 't_end'])
@apply_to_mask
@apply_to_audio
def subclip(self, t_start=0, t_end=None):
    ...
```

After adding **outplace** decorator
```
@convert_to_seconds(['t_start', 't_end'])
@apply_to_mask
@apply_to_audio
@outplace
def subclip(self, t_start=0, t_end=None):
    ...
```

I'm not sure if it is a legit fix or just co-incidence.

**Comment by TejasBob** [comment]
**Update:**

Adding the outplace operator isn't the right way solve the problem.

After going through source code for couple of times, I noticed that the steps used to close ffmpeg video reader are same as steps used to close ffmpeg audio reader except one step which is `proc.wait()`.

`close_proc()` method in readers.py doesn't have `proc.wait()`. I added it and ran the code. This time there were no defunct processes left behind.

I think it is a legit fix. I'm not sure if someone else has pointed this out earlier.

Thanks!
